src/components/Form.js

```diff
@@ -225,14 +225,14 @@ function Form(_props) {
     setState((state) => {
       const newErrors = Object.keys(fields.current).reduce((acc, name) => {
         if (typeof errorsMap[name] === "string") {
-          acc[name] = [errorsMap[name]];
-        } else if (Array.isArray(errorsMap[name])) {
-          acc[name] = errorsMap[name];
-        } else {
-          acc[name] = state.errors[name];
+          return setPath(acc, name, [errorsMap[name]]);
         }
-
-        return acc;
+        if (Array.isArray(errorsMap[name])) {
+          return setPath(acc, name, errorsMap[name]);
+        }
+        return getPath(state.errors, name)
+          ? setPath(acc, name, getPath(state.errors, name))
+          : acc;
```

Comment:
My understanding is that we want to preserve existing errors here. What if we kept `return acc;` here, but started the `reduce` from `{ ...state.errors }` (instead of `{}`). Would that achieve the same effect?